README.md

```diff
@@ -1 +1,253 @@
+Find the ***the sl_duty_cycle_config.h*** file as above, and enable EM2 mode and disable button as following.
+    ```c
+    #define DUTY_CYCLE_USE_LCD_BUTTON      0
+    #define DUTY_CYCLE_ALLOW_EM2           1
+    ```
+10. **Build project and flash to device**
+    - Click the hammer icon build the current project.
+    - After compiling is completed, find the binary file is located in **binaries** catalog, right click->**Flash to device...**->**choose the target device**->**OK**.    
+
+### Test the sleep current
+-  Build the project and flash the hex file to target board.
+-  Use the instrument to test sleep current. The sleep current as the following figure.  
+    <img src="images/sleep current.gif">  
+
+- Conclusion  
+The sleep current is **1.94uA** after disable TCXO and unused GPIO. Due to there still some GPIO is working, so the result is in accord with the data from datasheet.  
+
+## How to optomize the Rx on time
+In the sample project, the Rx on time is a fixed time even there is no any carrier in air. An appropriate approach is that radio will fast go to sleep when no any preamble is detected.
+### Modity the timing of Preamble detection 
+-  Add the macro to ***sl_duty_cycle_config.h*** file as following.
+    ```C
+    #define DUTY_CYCLE_PERIOD        (1500000)
+    #define DUTY_CYCLE_OFF_TIME      (DUTY_CYCLE_PERIOD - DUTY_CYCLE_ON_TIME)
+    ```
+-  Add the macro to ***app_init.c*** file as following.
+    ```C
+    // -----------------------------------------------------------------------------
+    //                              Macros and Typedefs
+    // -----------------------------------------------------------------------------
+    #define DUTY_CYCLY_SYNC_WORDS_TIME  1670
+    #define DUTY_CYCLY_MARGIN_TIME      1000
+    #define DUTY_CYCLY_SYNC_TIMEOUT     (DUTY_CYCLY_SYNC_WORDS_TIME + DUTY_CYCLE_PERIOD + DUTY_CYCLE_ON_TIME + DUTY_CYCLY_MARGIN_TIME)
+    ```
+-  Define the variable in ***app_init.c*** file and configurate the timing of duty cycle as following.
+    ```C
+    volatile RAIL_RxChannelHoppingConfigMultiMode_t duty_cycle_multi_mode_config = {
+        .timingSense = 1600,
+        .preambleSense = 2200,
+        .syncDetect = DUTY_CYCLY_SYNC_TIMEOUT,
+        .timingReSense = 1600,
+        .status = 0,
+    }; 
+    /// Config for the correct timing of the dutycycle API
+    RAIL_RxDutyCycleConfig_t duty_cycle_config = {
+      .delay = ((uint32_t) DUTY_CYCLE_OFF_TIME),
+      .delayMode = RAIL_RX_CHANNEL_HOPPING_DELAY_MODE_STATIC,
+      .mode = RAIL_RX_CHANNEL_HOPPING_MODE_MULTI_SENSE,
+      .parameter =  (uint32_t) (void *) &duty_cycle_multi_mode_config
+    //  .parameter = ((uint32_t) DUTY_CYCLE_ON_TIME)
+    };
+    ```
+-  Please find the ***sl_duty_utilicy.c*** file in the path ***gecko_sdk_3.2.1/app/flex/component/rail/sl_duty_cycle_core*** and modify the funcation of calculating preamble bit length as following. Please select the **Make a Copy** when you try to change this file.  
```

Comment:
-  Open the ***sl_duty_utilicy.c*** file in the folder ***gecko_sdk_3.2.2->app->flex->component->rail->sl_duty_cycle_core*** and modify the ***calculate_preamble_bit_length_from_time*** function as following. Please choose **Make a Copy** when you modify the file.